node/p2p/blossomsub.go

```diff
@@ -718,12 +733,20 @@ func initDHT(
 
 	reconnect()
 
+	bootstrapPeerIDs := make(map[peer.ID]struct{}, len(bootstrappers))
+	for _, peerinfo := range bootstrappers {
+		bootstrapPeerIDs[peerinfo.ID] = struct{}{}
+	}
 	go func() {
+	outer:
 		for {
 			time.Sleep(30 * time.Second)
-			if len(h.Network().Peers()) == 0 {
-				reconnect()
+			for _, p := range h.Network().Peers() {
+				if _, ok := bootstrapPeerIDs[p]; ok {
+					continue outer
```

Comment:
this looks like it'll loop until it's disconnected from bootstraps?

Reply:
The sleep on L744 still applies - it won't spin.

Reply:
Simplified without label jumping.